### PR TITLE
bump rust to 1.82

### PR DESCRIPTION
**Pull request by tcoratger**

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
lgtm pending @DaniPopes

File: crates/storage/provider/src/providers/static_file/manager.rs [hunk 1]
```diff
@@ -143,9 +143,9 @@ impl StaticFileProvider {
                         // appending/truncating rows
                         for segment in event.paths {
                             // Ensure it's a file with the .conf extension
-                            if !segment
+                            if segment
                                 .extension()
-                                .is_some_and(|s| s.to_str() == Some(CONFIG_FILE_EXTENSION))
+                                .is_none_or(|s| s.to_str() != Some(CONFIG_FILE_EXTENSION))
```
Comment (DaniPopes) [inline crates/storage/provider/src/providers/static_file/manager.rs:148]: This is more readable with the current is_some_and

Comment (tcoratger) [reply crates/storage/provider/src/providers/static_file/manager.rs:148]: yeah kind of agreed but this is clippy new suggestion... not sure this is good to put an allow() on this isolate case but if you have a strong opinion there I can revert :)

Comment (DaniPopes) [reply crates/storage/provider/src/providers/static_file/manager.rs:148]: yes please allow it

Comment (tcoratger) [reply crates/storage/provider/src/providers/static_file/manager.rs]: done

**Review by onbjerg — APPROVED** [review]

**Pull request merged**